src/client/components/input/partial-date.js

```diff
@@ -25,7 +25,17 @@ const ymdRegex = /^\d{4}-\d{2}-\d{2}$/;
 const ymRegex = /^\d{4}-\d{2}$/;
 const yRegex = /^\d{4}$/;
 
+/**
+* This class is derived from the React Component base class
+* to render the date field and check validity of the value filled in it.
```

Comment:
This actually renders a partial date - that is, it can render and validate any of these three formats: "YYYY", "YYYY-MM", "YYYY-MM-DD" - this allows the editor to be vague where information isn't know.